tracim/views/core_api/workspace_controller.py

```diff
@@ -0,0 +1,90 @@
+import typing
+
+from pyramid.config import Configurator
+from sqlalchemy.orm.exc import NoResultFound
+
+from tracim.lib.core.userworkspace import RoleApi
+from tracim.lib.utils.authorization import require_workspace_role
+from tracim.models.context_models import WorkspaceInContext
+from tracim.models.context_models import UserRoleWorkspaceInContext
+from tracim.models.data import UserRoleInWorkspace
+
+try:  # Python 3.5+
+    from http import HTTPStatus
+except ImportError:
+    from http import client as HTTPStatus
+
+from tracim import hapic, TracimRequest
+from tracim.exceptions import NotAuthentificated
+from tracim.exceptions import InsufficientUserProfile
+from tracim.exceptions import WorkspaceNotFound
+from tracim.lib.core.user import UserApi
+from tracim.lib.core.workspace import WorkspaceApi
+from tracim.views.controllers import Controller
+from tracim.views.core_api.schemas import WorkspaceSchema
+from tracim.views.core_api.schemas import UserSchema
+from tracim.views.core_api.schemas import WorkspaceIdPathSchema
+from tracim.views.core_api.schemas import WorkspaceMemberSchema
+
+class WorkspaceController(Controller):
+
+    @hapic.with_api_doc()
+    @hapic.handle_exception(NotAuthentificated, HTTPStatus.UNAUTHORIZED)
```

Comment:
which part of code generate the `NotAuthenticated` exception ?

Reply:
When request.current_user is called, the first time, authentication of user is automaticallty checked.

Reply:
hm... ok. by the way, rename `NotAuthentificated` to `NotAuthenticated`

Reply:
Done.